### ISSUE TITLE
Keras softmax function is not supported by tensorflow-gpu 1.4.1

### ISSUE THREAD

**Issue by SidJain1412** [tensorflow/tensorflow#20118]
I have a nvidia GeForce 940M which supports only CUDA 8.
CUDA 8 needs cudnn 7 to work.
CUDA 8 only supports tensorflow-gpu 1.4.1 and older.
tensorflow-gpu 1.4.1 does not support the Keras softmax function. (throws an error saying recieved extra positional argument)

My only option here seems to be to buy a new GPU which supports CUDA 9, please help.

**Comment by brge17** [comment]
https://github.com/keras-team/keras/issues/10440

```As a general rule, the latest release of Keras is compatible with the latest release of TensorFlow as well as the previous release of TensorFlow (in most cases, it is actually compatible with several prior TF releases, but that is not guaranteed).

TF 1.3 is now 5 releases behind (soon to be 6, since 1.9rc is already out), so you should either use a prior version of Keras or use an updated version of TensorFlow.```

**Comment by brge17** [comment]
I think you are misunderstanding the TensorFlow gpu requirements:

https://github.com/tensorflow/tensorflow/releases

```
TensorFlow 1.7 may be the last time we support Cuda versions below 8.0.
Starting with TensorFlow 1.8 release, 8.0 will be the minimum supported
version.
TensorFlow 1.7 may be the last time we support cuDNN versions below 6.0.
Starting with TensorFlow 1.8 release, 6.0 will be the minimum supported
version.
```

TensorFlow 1.8.0 works with CUDA 8, CUDNN 6

**Comment by SidJain1412** [comment]
@brge17 TensorflowGPU does not work with CUDA 8 CUDNN 7 or 6.
I cannot use another version of Keras either as they are not compatible with my versions either. I have triple checked.

**Comment by SidJain1412** [comment]
@ghostplant That seems like it would work, really appreciate your reply, but I do not understand how I can implement that. I'm kind of new to this, could you please help?

**Comment by ghostplant** [comment]
@SidJain1412 Do you use Ubuntu or Centos?

**Comment by SidJain1412** [comment]
@ghostplant Windows10   ':-)

**Comment by ghostplant** [comment]
@SidJain1412 Then I think you need to search the location of filename `tensorflow_backend.py` which should be within a certain directory of Python home path. Open it with an editor, when change the line below from:

```sh
    return tf.nn.softmax(x, axis=axis)
```

into:
```sh
    return tf.nn.softmax(x)    # Just remove the second argument will be okay
```

**Comment by SidJain1412** [comment]
@ghostplant God, thank you so much! Got another problem now though for one of my programs.

while_loop() got an unexpected keyword argument 'maximum_iterations'

This mentioned while_loop() is on line 2957 of the same backend.py file, and the code I have written is working on other systems with different versions of tensorflow. If you have some time could you please look at it? (I'm using LSTM to generate text and it needs to access the RNN part of the backend code, can't figure out why)

Thanks again!

**Comment by ghostplant** [comment]
@SidJain1412 Are you considering just use keras 2.1.5? Since it will be a lot easier to get over these incompatibilities.

```sh
pip install keras==2.1.5
```

**Comment by SidJain1412** [comment]
@ghostplant I am currently running 2.2.0. Will downgrading really help?

**Comment by ghostplant** [comment]
@SidJain1412 Yes, 2.1.5 is exactly for tensorflow 1.4.1.

**Comment by ghostplant** [comment]
If you prefer keras==2.2.0, you can continue to change the `tensorflow_backend.py` below from:

```sh
            swap_memory=True,
            maximum_iterations=input_length)
```
into:
```sh
            swap_memory=True)
```

**Comment by SidJain1412** [comment]
@ghostplant I didn't realise that keras>2.1.6 didn't support tensorflow<1.5 oh gosh everything's working now that I'm on keras 2.1.5 thank you so much man!
